vm/src/memory/constant_pool.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "errors.h"
+#include "memory/strings.h"
+#include <memory>
+#include <vector>
+
+class ByteReader;
+
+
+/**
+ * defines the type for an index inside the constant pool.
+ * */
+typedef uint32_t constant_index;
+
+/**
+ * Kind of constant that can be contained in the constant pool
+ * */
+enum PoolConstantType {
+    /// an utf-8 string
+    C_STR,
```

Comment:
```suggestion
    UTF8_STR,
```
The string isn't encoded as a C string.